src/rust/engine/fs/Cargo.toml

```diff
@@ -22,6 +22,7 @@ lazy_static = { workspace = true }
 log = { workspace = true }
 parking_lot = { workspace = true }
 protos = { path = "../protos" }
+pyo3 = { workspace = true }
```

Comment:
Yeah, this is also what we do for the native options parser. There is some repetition, but it keeps the Rust side free of pyo3. This is especially important if we end up using the `fs` code in other rust code.

Reply:
It seems to me like this would not work for enums though because annotating an outer `PyPathMetadataLind`  type with `#[pyclass]` would not generate the enum variants visible in Python which are generated currently by pyo3 when the `pyclass` annoation is directly on the `PathMetadataKind`. 

pyo3 does not appear to have an equivalent to the `serde` crate's `#[serde(flatten)]` attribute which removes a layer of indirection with nested types.

I could define the enumeration in Python instead of it being a Rust type or expose some other type. Indeed, an earlier iteration of this PR did in fact expose the enumeration as a string, but the earlier review comments asked me to use pyo3's enum functionality ...

So I can restore my earlier approach but we should decide on _some_ acceptable approach to avoid development going in circles.

Reply:
For the enum, I'd suggest going for the duplication approach.

It's annoying, but if we use (exhaustive) `match` to convert between them python version and Rust version, it's not error prone: adding a variant in one place will trigger a compiler error at the appropriate conversion function, to remind us to update the other place.

Reply:
Yeah, I am fine with some duplication/boilerplate, since this will be changed rarely, if ever, and even then only by people who really know what they're doing.